lfs/endpoint.go

```diff
@@ -37,21 +36,95 @@ func NewEndpointFromCloneURL(url string) Endpoint {
 }
 
 // NewEndpoint initializes a new Endpoint for a given URL.
-func NewEndpoint(url string) Endpoint {
-	e := Endpoint{Url: url}
+func NewEndpoint(rawurl string) Endpoint {
+	var endpoint Endpoint
+	// Check for SSH URLs
+	// Support ssh://user@host.com/path/to/repo and user@host.com:path/to/repo
+	u, err := url.Parse(rawurl)
+	if err == nil {
+		u = processBareSshUrl(u)
+		switch u.Scheme {
+		case "ssh":
+			endpoint = endpointFromSshUrl(u)
+		case "http", "https":
+			endpoint = endpointFromHttpUrl(u)
+		default:
+			// Just passthrough to preserve
+			endpoint = Endpoint{Url: rawurl}
+		}
+	}
```

Comment:
Should we really return an empty `Endpoint` in case of `err != nil`?  Why not `Endpoint{Url: rawurl}`?